pkg/log/logshipper/logshipper.go

```diff
@@ -0,0 +1,127 @@
+package logshipper
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/kolide/launcher/pkg/agent/types"
+	"github.com/kolide/launcher/pkg/sendbuffer"
+)
+
+// TODO: consolidate this var, but where?
+var observabilityIngestTokenKey = []byte("observability_ingest_auth_token")
+
+const (
+	truncatedFormatString = "%s[TRUNCATED]"
+)
+
+type LogShipper struct {
+	sender            *authedHttpSender
+	sendBuffer        *sendbuffer.SendBuffer
+	logger            log.Logger
+	knapsack          types.Knapsack
+	stopFunc          context.CancelFunc
+	isShippingEnabled bool
+}
+
+func New(k types.Knapsack) (*LogShipper, error) {
+	token, _ := k.TokenStore().Get(observabilityIngestTokenKey)
+
+	logEndpoint, err := logEndpoint(k)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I do think we should check the endpoint for validity, but with how this is set up right now I'm not sure we should be returning an error here. If we start off with an invalid/malformed URL, the control server will never be able to update it to a valid one. Maybe there's another/later place we could parse and check the endpoint, or maybe we could log + ignore the error here?

Reply:
doing this now, log and ignore, led to some consolidation of is enabled logic